nibabel/streamlines/array_sequence.py

```diff
@@ -118,7 +148,39 @@ def total_nb_rows(self):
     @property
     def data(self):
         """ Elements in this array sequence. """
-        return self._data
+        warnings.warn("The 'ArraySequence.data' property has been deprecated"
+                      " in favor of 'ArraySequence.get_data()'.",
+                      DeprecationWarning,
+                      stacklevel=2)
+        return self.get_data()
+
+    def get_data(self):
```

Comment:
Are there cases where someone might want read-only access to the original data, without performing a copy?

Reply:
Since it's not always possible to return a view on *just* the data belonging to a sliced/indexed `ArraySequence`, I decided to always make it return a copy.

I know it is bad practice but I'm expecting Dipy's methods to deal directly with `._data` using `._offsets` and `._lengths`.

That said, maybe there exists a solution but I fail to see it. I'm happy to consider alternatives.

Reply:
Agree, I do no think that a lot of users might want a read-only access to the original data